lib/dynflow/executors/parallel/pool.rb

```diff
@@ -46,12 +46,17 @@ def delete(execution_plan_id)
           end
         end
 
-        def initialize(core, name, pool_size, transaction_adapter)
+        def initialize(world, core, name, pool_size, transaction_adapter)
+          @world = world
           @name = name
           @executor_core = core
           @pool_size     = pool_size
-          @free_workers  = Array.new(pool_size) { |i| Worker.spawn("worker-#{i}", reference, transaction_adapter) }
           @jobs          = JobStorage.new
+          @free_workers  = Array.new(pool_size) do |i|
+            name = "worker-#{i}"
+            telemetry_options = { :world => @world.id, :worker => name, :queue => @name }
```

Comment:
This should use `telemetry_options` method as a base, shouldn't it?

Reply:
Good catch